app/src/main/java/com/github/sdp/ratemyepfl/adapter/ReviewableAdapter.kt

```diff
@@ -0,0 +1,136 @@
+package com.github.sdp.ratemyepfl.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.Filter
+import android.widget.Filterable
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.model.items.Course
+import com.github.sdp.ratemyepfl.model.items.Reviewable
+
+class ReviewableAdapter(private val onClick: (Reviewable) -> Unit) :
+    ListAdapter<Reviewable, ReviewableAdapter.ReviewableViewHolder>(ReviewableDiffCallback),
+    Filterable {
+
+    private var list = mutableListOf<Reviewable>()
+
+    inner class ReviewableViewHolder(reviewableView: View) :
+        RecyclerView.ViewHolder(reviewableView) {
+
+        private val reviewableTextView: TextView = reviewableView.findViewById(R.id.reviewableId)
+        private val reviewBut: Button = reviewableView.findViewById(R.id.reviewableButton)
+        private var currentReviewable: Reviewable? = null
+
+        init {
+            reviewBut.setOnClickListener {
+                currentReviewable?.let {
+                    onClick(it)
+                }
+            }
+        }
+
+        /* Bind room id. */
+        fun bind(reviewable: Reviewable) {
+            currentReviewable = reviewable
+            reviewableTextView.text = reviewable.toString()
+        }
+    }
+
+    /* Creates and inflates view and return RoomViewHolder. */
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReviewableViewHolder {
+        return ReviewableViewHolder(
+            LayoutInflater.from(parent.context).inflate(
+                R.layout.reviewable_item,
+                parent,
+                false
+            )
+        )
+    }
+
+    /* Gets current room and uses it to bind view. */
+    override fun onBindViewHolder(holder: ReviewableViewHolder, position: Int) {
+        val room = getItem(position)
+        holder.bind(room)
+    }
+
+    fun setData(list: MutableList<Reviewable>?) {
+        this.list = list!!
+        submitList(list)
+    }
+
+    override fun getFilter(): Filter {
+        return reviewableSearchFilter
+    }
+
+    fun getFilterMethod(filter: (CharSequence?) -> List<Reviewable>): Filter {
```

Comment:
As `reviewableSearchFilter` is private, it would make more sense to make this method private too

Reply:
Yes totally, I will correct it.